core/src/test/java/org/apache/iceberg/TestSplitPlanning.java

```diff
@@ -218,6 +237,17 @@ public void testSplitPlanningWithOffsets() {
         "We should get one task per row group", 32, Iterables.size(scan.planTasks()));
   }
 
+  @Test
+  public void testAdaptiveSplitPlanningWithOffests() {
+    table.updateProperties().set(TableProperties.ADAPTIVE_SPLIT_PLANNING, "true").commit();
```

Comment:
This pathway needs a more test imho, we mostly deal with files with offsets so we should make sure we aren't doing anything weird in those cases. We also have a adaptive function above which has a lot of early exits and behaviors so we should make sure all of those exits are tested.